jdaviz/configs/mosviz/helper.py

```diff
@@ -46,89 +55,133 @@ def _extend_world(self, spec1d, ext):
         world = np.hstack((prepend, world, append))
         return world
 
-    def _update_spec2d_x_axis(self, change):
+    def _update_spec2d_x_axis(self, _):
         # This assumes the two spectrum viewers have the same x-axis shape and
         # wavelength solution, which should always hold
-        if change['old'] is None:
-            pass
-        else:
-            name = change['name']
-            if name not in ['min', 'max']:
-                return
-            new_val = change['new']
-            spec1d = self.app.get_viewer('table-viewer')._selected_data["spectrum-viewer"]
-            extend_by = int(self.app.data_collection[spec1d]["World 0"].shape[0])
-            world = self._extend_world(spec1d, extend_by)
-
-            # Warn the user if they've panned far enough away from the data
-            # that the viewers will desync
-            if new_val > world[-1] or new_val < world[0]:
-                msg = "Warning: panning too far away from the data may desync\
-                      the 1D and 2D spectrum viewers"
-                msg = SnackbarMessage(msg, color='warning', sender=self)
-                self.app.hub.broadcast(msg)
-
-            idx = float((np.abs(world - new_val)).argmin()) - extend_by
-            scales = self.app.get_viewer('spectrum-2d-viewer').scales
-            old_idx = getattr(scales['x'], name)
-            if idx != old_idx:
-                setattr(scales['x'], name, idx)
-
-    def _update_spec1d_x_axis(self, change):
+        table_viewer = self.app.get_viewer('table-viewer')
+
+        if self._update_in_progress or table_viewer.row_selection_in_progress:
+            return
+
+        min_1d = self._scales1d.min
+        max_1d = self._scales1d.max
+
+        spec1d = table_viewer._selected_data["spectrum-viewer"]
+        extend_by = int(self.app.data_collection[spec1d]["World 0"].shape[0])
+        world = self._extend_world(spec1d, extend_by)
+
+        # Workaround for flipped data
+        min_world, max_world = (world[0], world[-1]) if not self._is_world_flipped()\
+            else (world[-1], world[0])
+
+        # Warn the user if they've panned far enough away from the data
+        # that the viewers will desync
+        if min_1d < min_world or max_1d > max_world:
```

Comment:
In Python convention, the max is usually not inclusive. But I am unfamiliar with this code, so someone should double check.
```suggestion
        if min_1d < min_world or max_1d >= max_world:
```

Reply:
`max_world` is the value of the last world element, so we do want `>` here.